.github/workflows/build.yaml

```
@@ -23,8 +20,60 @@ env:
+            git add .
+            git commit -m "style: auto format with clang-format."
+            git push
+            echo "Commit format."
+          else
+            echo "No formatting changes were necessary."
+          fi
+
   build:
-    if: github.event.pull_request.draft == false
+    needs: format
+    if: |
```

actionlint (1.7.4)

undefined variable "ithub". available variables are "env", "github", "inputs", "job", "matrix", "needs", "runner", "secrets", "steps", "strategy", "vars"

(expression)